### ISSUE TITLE
Dev tools / Inspect element window doesn't open on latest brave releases (VERY slow to open)

### ISSUE THREAD

**Issue by shanytc** [brave/brave-browser#14721]
## Description 
On my MacBook Pro (13-inch, 2020),
The latest brave browser releases completely broke the inspect element window. It will no longer open immediately, and if it does, it's after 30 seconds or 1 minute sometimes not at all. When it does open it's unresponsive.

## Steps to Reproduce

   1.  Install Latest brave releases for OSX
   2.  Open any page
   3.  right click -> inspect element -> it's not showing / takes around 30secs to 1 minute to open

## Actual result:
Opens after 30secs or 1 minute like this: https://ibb.co/HtMLgGg

## Expected result:
Should open immediately like this: https://ibb.co/tCBHcMy

## Reproduces how often: 
100% of the time.

## Brave version (brave://version info)
Version 1.21.76

## Version/Channel Information:
Version 1.21.76 Chromium: 89.0.4389.86 (Official Build) (x86_64)

- Can you reproduce this issue with the current release?  yes
- Can you reproduce this issue with the beta channel?  yes
- Can you reproduce this issue with the nightly channel?  yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? no
- Does the issue resolve itself when disabling Brave Rewards? no
- Is the issue reproducible on the latest version of Chrome?  no

## Miscellaneous Information:
unresponsive behaviour for developers.

**Comment by jowsey** [comment]
Also get the same issue on 1.24.8 on Windows 10

**Comment by shanytc** [comment]
> Also get the same issue on 1.24.8 on Windows 10

Good to know! Seems this is a wide issue.

**Comment by bsclifton** [comment]
Fixed in 1.22.x with https://github.com/brave/brave-core/commit/f20e8ba35f798ee3a8206512705d7b708e67c090

This is still a problem in other channels though ☹️

**Comment by kjozwiak** [comment]
As per the above, we reverted the change that caused the issue in `1.22.66`. The fix will be done via https://github.com/brave/brave-browser/issues/14880.

**Comment by kjozwiak** [comment]
@brave/legacy_qa we'll basically need to make sure that the DevTools is working as expected on a default install. The fix that will address the underlying problem will be done via https://github.com/brave/brave-browser/issues/14880.

As per @darkdh, this was due to DevTools calls running through DoH.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Ubuntu 20.04 x64` using the following build:

```
Brave | 1.22.67 Chromium: 89.0.4389.90 (Official Build) (64-bit)
--- | ---
Revision | 62eb262cdaae9ef819aadd778193781455ec7a49-refs/branch-heads/4389@{#1534}
OS | Linux
```

* ensured that launching `DevTools` worked as expected via `Hamburger Menu` -> `More tools` --> `Developer tools`
* ensured that `CTRL + SHIFT + I` launched the DevTools without any issues

**Comment by rebron** [comment]
Looks good on `macOS 11.2.2` using the following build:

Brave | 1.22.67 Chromium: 89.0.4389.90 (Official Build) (arm64)
-- | --
Revision | 62eb262cdaae9ef819aadd778193781455ec7a49-refs/branch-heads/4389@{#1534}
OS | macOS Version 11.2.2 (Build 20D80)

- Ensured right click Inspect on web page opens up dev tools
- From hamburger menu -> Developer Tools opens and closes with no issue

Wasn't able to reproduce this issue on macOS 1.22.66 arm64 builds.

**Comment by shanytc** [comment]
> 1.22.67

Issue still persists in this build.
Developer tools will not open on a fresh new install.

**Comment by kjozwiak** [comment]
> > 1.22.67
> 
> Issue still persists in this build.
> Developer tools will not open on a fresh new install.

@shanytc have you edited the DoH settings under `brave://settings/security` by any chance? Example:

![image](https://user-images.githubusercontent.com/2602313/112228355-90630400-8c07-11eb-919b-c7780ce3dfbf.png)

We've checked every platform using `1.22.67` and it seems to be working in clean installs. When you go into `brave://version`, are you seeing `1.22.67` or `1.22.66`, Also, which platform are you using?

**Comment by shanytc** [comment]
> > > 1.22.67
> > 
> > 
> > Issue still persists in this build.
> > Developer tools will not open on a fresh new install.
> 
> @shanytc have you edited the DoH settings under `brave://settings/security` by any chance? Example:
> 
> ![image](https://user-images.githubusercontent.com/2602313/112228355-90630400-8c07-11eb-919b-c7780ce3dfbf.png)
> 
> We've checked every platform using `1.22.67` and it seems to be working. When you go into `brave://version`, are you seeing `1.22.67` or `1.22.66`, Also, which platform are you using? Sounds like OSX? Can you let us know the exact version you're on?

Here is the version information I'm running:

`
Brave | 1.22.67 Chromium: 89.0.4389.90 (Official Build) (x86_64)
-- | --
Revision | 62eb262cdaae9ef819aadd778193781455ec7a49-refs/branch-heads/4389@{#1534}
OS | macOS Version 10.15.7 (Build 19H524)
JavaScript | V8 8.9.255.20

`

I have the same DoH settings under brave://settings/security as you posted in the image.

**Comment by stephendonner** [comment]
Verified `PASSED` using 

Brave | 1.22.67 Chromium: 89.0.4389.90 (Official Build) (x86_64)
-- | --
Revision | 62eb262cdaae9ef819aadd778193781455ec7a49-refs/branch-heads/4389@{#1534}
OS | macOS Version 11.2.3 (Build 20D91)

Tested and verified:

1. Right-click -> `Inspect` opens the `Developer Tools` promptly, and selects the chosen element
2. Keybinding `Option + Command + I` opens (and toggles closed) the `Developer Tools` panel promptly
3. Hamburger menu -> `More Tools` -> `Developer Tools` opens them promptly

**Comment by riequilibrium** [comment]
Hello,
I updated Brave to the latest version but the developer tools are still loading very slowly.
Having this issue for a while now.

macOS Catalina 10.15.7 (19H524)
Brave Version 1.22.67 Chromium: 89.0.4389.90 (Official build) (x86_64)

**Comment by shanytc** [comment]
I have not idea who tested the fix, but clearly it hasn't been fixed.
Also, why would you every want to do DNS check for opening a developer tools...?

**Comment by riequilibrium** [comment]
It seems that the built-in adblocker is intercepting the devtool call, disabling the Secure DNS or using another DNS like Cloudflare actually fixed the slowdown, but it is clearly a temporary workaround.

You can try it, go to Settings->Security, under Advanced you'll find the option for Secure DNS, select Cloudflare and now the devtools should open instantly!

![image](https://user-images.githubusercontent.com/31438507/112293363-5b4fc380-8c92-11eb-854c-fdf51baed658.png)

Hopes this helps you!

**Comment by M-BenAli** [comment]
Relaunching brave fixed the slow developer tools startup, this option should ONLY be available if the update hasn't fully been completed yet. Go to: **Settings -> About brave ->Relaunch brave option**.

**Comment by shanytc** [comment]
> Relaunching brave fixed the slow developer tools startup for me, this option is ONLY be available if the update hasn't fully been completed yet. Go to: **Settings -> About brave ->Relaunch brave option**.

Doesn't fix the issue.

**Comment by jowsey** [comment]
> It seems that the built-in adblocker is intercepting the devtool call, disabling the Secure DNS or using another DNS like Cloudflare actually fixed the slowdown, but it is clearly a temporary workaround.

Can confirm this works as a fix for now, using Cloudflare DNS

**Comment by shanytc** [comment]
> > It seems that the built-in adblocker is intercepting the devtool call, disabling the Secure DNS or using another DNS like Cloudflare actually fixed the slowdown, but it is clearly a temporary workaround.
> 
> Can confirm this works as a fix for now, using Cloudflare DNS

It still takes around 10+ seconds for the developer tools to show up with CloudFlare DNS... this is a hack and not a solution.
Developer tools should not even use any dns remote check.

**Comment by dougfarnes** [comment]
This is not completely fixed for my install either.

@kjozwiak, my settings are as shown in your screenshot and dev tools take forever to open right now.  However, if I apply the workaround by @riequilibrium (forcing Cloudflare DNS), dev tools open instantly.  Also, if I connect to a self-hosted Wireguard VPN, dev tools load without delay.

Not sure what this means, if my local DNS setup is conflicting with what Brave is trying to do vis-a-vis CNAME adblocking, or how that relates to dev tools, but wanted to add another data point.

**Comment by bsclifton** [comment]
Thanks for the reports folks 😄 We do have a proper fix for this which I'd like to ask if you can try

This would be on Nightly - but you can download / install / test and then uninstall after testing. We just need to know it fixes for sites you visit. The version you can download is here:
https://github.com/brave/brave-browser/releases/tag/v1.24.18

Windows (64-bit) users will want to download this one:
https://github.com/brave/brave-browser/releases/download/v1.24.18/BraveBrowserStandaloneNightlySetup.exe

macOS users will want to download this one:
https://github.com/brave/brave-browser/releases/download/v1.24.18/Brave-Browser-Nightly-x64.dmg

cc: @dougfarnes @shanytc @jowsey @riequilibrium

**Comment by shanytc** [comment]
> Thanks for the reports folks 😄 We do have a proper fix for this which I'd like to ask if you can try
> 
> This would be on Nightly - but you can download / install / test and then uninstall after testing. We just need to know it fixes for sites you visit. The version you can download is here:
> https://github.com/brave/brave-browser/releases/tag/v1.24.18
> 
> Windows (64-bit) users will want to download this one:
> https://github.com/brave/brave-browser/releases/download/v1.24.18/BraveBrowserStandaloneNightlySetup.exe
> 
> macOS users will want to download this one:
> https://github.com/brave/brave-browser/releases/download/v1.24.18/Brave-Browser-Nightly-x64.dmg
> 
> cc: @dougfarnes @shanytc @jowsey @riequilibrium
> 
> For those interested, the fix can be found here:
> [brave/brave-core#8336](https://github.com/brave/brave-core/pull/8336)

Nightly works perfectly!!!

**Comment by kjozwiak** [comment]
Thanks @shanytc 👍 We'll get a HF out tomorrow with the above fix on release.

**Comment by shanytc** [comment]
> Thanks @shanytc 👍 We'll get a HF out tomorrow with the above fix on release.

Good work, thanks guys!

**Comment by riequilibrium** [comment]
> Thanks for the reports folks 😄 We do have a proper fix for this which I'd like to ask if you can try
> 
> This would be on Nightly - but you can download / install / test and then uninstall after testing. We just need to know it fixes for sites you visit. The version you can download is here:
> https://github.com/brave/brave-browser/releases/tag/v1.24.18
> 
> Windows (64-bit) users will want to download this one:
> https://github.com/brave/brave-browser/releases/download/v1.24.18/BraveBrowserStandaloneNightlySetup.exe
> 
> macOS users will want to download this one:
> https://github.com/brave/brave-browser/releases/download/v1.24.18/Brave-Browser-Nightly-x64.dmg
> 
> cc: @dougfarnes @shanytc @jowsey @riequilibrium
> 
> For those interested, the fix can be found here:
> [brave/brave-core#8336](https://github.com/brave/brave-core/pull/8336)

I can confirm that the Nightly version works like a charm!
Looking forward to install the hotfix on the stable release,
Thanks a lot!

**Comment by garyburgmann** [comment]
This has been driving me bonkers! But at least this issue has lead me to discover `raindrop.io` while I run Firefox :)

Can confirm that the Nightly version appears to also work for me running `macOS Version 11.2.3 (Build 20D91)`

**Comment by bsclifton** [comment]
@garyburgmann if you update the Release channel version, it should now be fixed too 😄

**Comment by amuagarwal** [comment]
It look like that the issue is resurfaced with [Version 1.45.116 Chromium: 107.0.5304.91 (Official Build) (64-bit)]

**Comment by stephendonner** [comment]
> It look like that the issue is resurfaced with [Version 1.45.116 Chromium: 107.0.5304.91 (Official Build) (64-bit)]

Please file a new issue with specific details (steps to reproduce, any performance timings, etc.  - as much as you can).  Thanks.

**Comment by amuagarwal** [comment]
> > It look like that the issue is resurfaced with [Version 1.45.116 Chromium: 107.0.5304.91 (Official Build) (64-bit)]
> 
> Please file a new issue with specific details (steps to reproduce, any performance timings, etc. - as much as you can). Thanks.

Already filled issue, but seems it fixed with [Version 1.45.118 Chromium: 107.0.5304.91 (Official Build) (64-bit)]
[#6469](https://github.com/brave/brave-browser/issues/26469)
